<?xml version="1.0"?>
<launch>

    <!--<arg name="model" default="standalone"/>-->
    <arg name="first_tb3"  default="tb3_1"/>

    <arg name="first_tb3_x_pos" default="0.0"/>
    <arg name="first_tb3_y_pos" default="0.0"/>
    <arg name="first_tb3_z_pos" default="0.0"/>
    <arg name="first_tb3_yaw"   default="0.0"/>

    <node pkg="ros_vrpn_client" name="$(arg first_tb3)" type="ros_vrpn_client" args="_vrpn_server_ip:=192.168.1.104" required="true" output="screen"/>

    <include file = "$(find turtlebot_bringup)/launch/minimal.launch" output = "screen">
    </include>

    <group ns = "$(arg first_tb3)">
        <arg name="waypoints_file" default="$(find setpoint_generator)/waypoint_files/basic_waypoints.yaml" />
        <node name="setpoint_publisher" pkg="setpoint_generator" type="get_setpoint_node.py" clear_params="true" output="screen">
            <param name="name" value="$(arg first_tb3)" type="str"/>
            <remap from="pose_stamped" to="/$(arg first_tb3)/nwu/pose_stamped"/>
            <remap from="setpoint" to="/$(arg first_tb3)/target_pose"/>
            <rosparam file="$(arg waypoints_file)"/>
            <rosparam command="load" file="$(find setpoint_generator)/params/basic_param.yaml"/>
        </node>

        <node pkg="rospy_turtlebot_pid" name="pid_controller" type="controller_node.py" output="screen">
            <remap from="agent_pose" to="/$(arg first_tb3)/nwu/pose_stamped" />
            <remap from="target_pose" to="/$(arg first_tb3)/target_pose" />
            <remap from="pub_info" to="/mobile_base/commands/velocity"/>
            <rosparam file="$(find rospy_turtlebot_pid)/config/params.yaml" />
        </node>
    </group>
</launch>
